files/scripts/healdata/heal-cedar-data-ingest.py

```diff
@@ -79,53 +79,84 @@ def update_filter_metadata(metadata_to_update):
-        sys.exit(1)
-
-    print(f"Successfully got {len(metadata_return['metadata'])} record(s) from CEDAR directory")
-    for cedar_record in metadata_return["metadata"]:
-        if "appl_id" not in cedar_record:
-            print("This record doesn't have appl_id, skipping...")
-            continue
-        cedar_record_id = str(cedar_record["appl_id"])
-
-        # Get the metadata record for the nih_application_id
-        mds = requests.get(f"http://revproxy-service/mds/metadata/{cedar_record_id}",
-            headers=token_header
-        )
-        if mds.status_code == 200:
-            mds_res = mds.json()
-            mds_cedar_register_data_body = {}
-            mds_discovery_data_body = {}
-            if mds_res["_guid_type"] == "discovery_metadata":
-                print("Metadata is already registered. Updating MDS record")
-            elif mds_res["_guid_type"] == "unregistered_discovery_metadata":
-                print("Metadata is has not been registered. Registering it in MDS record")
+limit = 10
+offset = 0
+
+# initalize this to be bigger than our inital call so we can go through while loop
+total = 100
+
+while((limit + offset <= total)):
+    # Get the metadata from cedar to register
+    print("Querying CEDAR...")
+    cedar = requests.get(f"http://revproxy-service/cedar/get-instance-by-directory-offset/{dir_id}?limit={limit}&offset={offset}", headers=token_header)
+
+    # If we get metadata back now register with MDS
+    if cedar.status_code == 200:
+        metadata_return = cedar.json()
+        if "metadata" not in metadata_return:
+            print("Got 200 from CEDAR wrapper but no metadata in body, something is not right!")
+            sys.exit(1)
+
+        total = metadata_return["metadata"]["totalCount"]
+        returned_records = len(metadata_return["metadata"]["records"])
+        print(f"Successfully got {returned_records} record(s) from CEDAR directory")
+        for cedar_record in metadata_return["metadata"]["records"]:
+            if "appl_id" not in cedar_record:
+                print("This record doesn't have appl_id, skipping...")
                 continue
-            pydash.merge(mds_discovery_data_body, mds_res["gen3_discovery"], cedar_record)
-            mds_discovery_data_body = update_filter_metadata(mds_discovery_data_body)
-            mds_cedar_register_data_body["gen3_discovery"] = mds_discovery_data_body
-            mds_cedar_register_data_body["_guid_type"] = "discovery_metadata"
-
-            print("Metadata is now being registered.")
-            mds_put = requests.put(f"http://revproxy-service/mds/metadata/{cedar_record_id}",
-                headers=token_header,
-                json = mds_cedar_register_data_body
+
+            cedar_appl_id = str(cedar_record["appl_id"])
+            # have to query the mds to find the cedar record id
+            query = requests.get(f"http://revproxy-service/mds/metadata?gen3_discovery.appl_id={cedar_appl_id}")
```

Comment:
can we use something like `query = requests.get(f"http://revproxy-service/mds/metadata?data=True&gen3_discovery.appl_id={cedar_appl_id}")` to get the entrie metadata block back via one single query rather than query MDS twice?

Reply:
Added this to the script